src/Text/Pandoc/Writers/LaTeX.hs

```diff
@@ -1472,18 +1483,36 @@ citationsToBiblatex (c:cs)
                     NormalCitation -> "\\autocite"
       return $ text cmd <>
                braces (literal (T.intercalate "," (map citationId (c:cs))))
-  | otherwise = do
-    let cmd = case citationMode c of
+  | otherwise 
+    = do  
+      let cmd = case citationMode c of
                     SuppressAuthor -> "\\autocites*"
                     AuthorInText   -> "\\textcites"
                     NormalCitation -> "\\autocites"
-    let convertOne Citation { citationId = k
-                            , citationPrefix = p
-                            , citationSuffix = s
-                            }
-                = citeArguments p s k
-    args <- mapM convertOne (c:cs)
-    return $ text cmd <> foldl' (<>) empty args
+
+      let groups = map correctOrder $ reverse $ foldl' grouper [([[]], [])] (c:cs)
+      docGroups <- mapM citeArgsList groups      
+
+      return $ text cmd <> (mconcat docGroups)
+
+  where grouper prev cit = grouper' (citationPrefix cit)
+               (citationSuffix cit) (citationId cit) 
+            where takefst = (fst =<< take 1 prev)  
+                  grouper' [] sfx cid
+                     | null sfx = addToGroup id 
+                     | otherwise = addToGroup ((:) sfx) 
+                     where addToGroup fn 
+                             | (1 < (length takefst)) = 
+                                    (fn $ drop 1 $ takefst, [cid]):prev
+                             | otherwise = (\(a, b) -> (fn a, cid:b):drop 1 prev) 
+                                    =<< take 1 prev
+                  grouper' pfx [] cid = ([pfx], [cid]):prev
+                  grouper' pfx sfx cid = ([sfx, pfx], [cid]):prev 
+
+        correctOrder (pfxs, ids) = (reverse pfxs, reverse ids)
+        citeArgsList ((p:s:_), k) = citeArgumentsList p s k
+        citeArgsList ((p:_), k) = citeArgumentsList p [] k
+        citeArgsList ([], k) = citeArgumentsList [] [] k
 
```

Comment:
This looks like it's probably correct, but I think the code could be clearer (and then it would be more obvious whether it's right).   For example, I think it would be clearer if you used pattern matching for `addToGroup` instead of `take 1` and then `drop 1`.  Something like
```
case prev of
  ((a,b):rest
    | null a ->
    | otherwise ->
  [] -> 
```
This would avoid the need for the confusing `takefst` stuff and make it clearer what's going on.